### PR TITLE
TEPP Updates

### PR DESCRIPTION
**Pull request by potatoqualitee**
-- DO NOT MERGE until @FriedrichWeinmann has reviewed ---

### CODE REVIEW

File: internal/dynamicparams/job.ps1 [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	
+	$server = $fakeBoundParameter['SqlInstance']
+	
+	if (-not $server) {
+		$server = $fakeBoundParameter['Source']
+	}
+	
+	if (-not $server) {
+		$server = $fakeBoundParameter['SqlInstance']
+	}
```
Comment (FriedrichWeinmann) [inline internal/dynamicparams/job.ps1]: Isn't that redundant?

File: internal/dynamicparams/database.ps1 [hunk 2]
```diff
@@ -70,4 +70,4 @@ $ScriptBlock = {
 }
 
 Register-DbaTeppScriptblock -ScriptBlock $ScriptBlock -Name Database
-Register-DbaTeppScriptblock -ScriptBlock $ScriptBlock -Name Exclude
\ No newline at end of file
+Register-DbaTeppScriptblock -ScriptBlock $ScriptBlock -Name ExcludeDatabase
```
Comment (FriedrichWeinmann) [inline internal/dynamicparams/database.ps1]: Scriptblocks need only be registered once and are named for the kind of objects they complete, not what is being done with them afterwards, so kill the second line, it's not being used.

File: internal/dynamicparams/login.ps1 [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	
+	$server = $fakeBoundParameter['SqlInstance']
+	
+	if (-not $server) {
+		$server = $fakeBoundParameter['Source']
+	}
+	
+	if (-not $server) {
+		$server = $fakeBoundParameter['SqlInstance']
+	}
```
Comment (FriedrichWeinmann) [inline internal/dynamicparams/login.ps1]: Again: Redundant.
A relic from bulk-renaming things?

File: internal/dynamicparams/login.ps1 [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+		return
+	}
+	catch {
+		[Sqlcollective.Dbatools.TabExpansion.TabExpansionHost]::Scripts["login"].LastDuration = (Get-Date) - $start
+		return
+	}
+}
+
+Register-DbaTeppScriptblock -ScriptBlock $ScriptBlock -Name Login
+Register-DbaTeppScriptblock -ScriptBlock $ScriptBlock -Name ExcludeLogin
```
Comment (FriedrichWeinmann) [inline internal/dynamicparams/login.ps1]: Scriptblocks need only be registered once and are named for the kind of objects they complete, not what is being done with them afterwards, so kill the second line, it's not being used.

File: internal/dynamicparams/login.ps1 (hunk 3 above)
Comment (potatoqualitee) [inline internal/dynamicparams/login.ps1]: Yes

File: internal/dynamicparams/login.ps1 (hunk 4 above)
Comment (potatoqualitee) [inline internal/dynamicparams/login.ps1]: k!

File: internal/dynamicparams/database.ps1 (hunk 2 above)
Comment (potatoqualitee) [inline internal/dynamicparams/database.ps1]: this one was a fix, ill kill the whole line

**Comment by potatoqualitee** [comment]
Thank you 👍 Fixed and ready to merge.

Is this the direction we'd like to go? Cache all the things we are auto loading? btw, I did some tests and was 😻 

image

**Comment by potatoqualitee** [comment]
💃

**Comment by FriedrichWeinmann** [comment]
No, we don't actually want to auto-cache everything automatically:
Retrieving the data each time slows the connects a bit. Won't be noticable with the current amount, but can grow if we really _do_ cache everything.

Plan is to auto-cache the most frequent stuff and only start caching the other data after its respective TEPP was first requested.
Which is where my next upgrade plan for the TEPP Stuff will come in, but that's still a bit in the future (UG meeting tomorrow and test troubleshooting just come first)

**Comment by potatoqualitee** [comment]
excellent! do i just go forward as is with this plan?

**Comment by FriedrichWeinmann** [comment]
You just go forward with the current style, I'll be able to do a global refactor when I add the new style with very little effort.

**Comment by potatoqualitee** [comment]
👍  /cc @wsmelton

**Pull request merged**